pkg/controller/controller.go

```diff
@@ -717,24 +864,19 @@ func (p *csiProvisioner) volumeHandleToId(handle string) string {
 // - ${pvc.namespace}
 //
 // an error is returned in the following situations:
-// - only one of name or namespace is provided
-// - the name or namespace parameter contains a token that cannot be resolved
+// - the nameTemplate or namespaceTemplate contains a token that cannot be resolved
 // - the resolved name is not a valid secret name
 // - the resolved namespace is not a valid namespace name
-func getSecretReference(nameKey, namespaceKey string, storageClassParams map[string]string, pvName string, pvc *v1.PersistentVolumeClaim) (*v1.SecretReference, error) {
-	nameTemplate, hasName := storageClassParams[nameKey]
-	namespaceTemplate, hasNamespace := storageClassParams[namespaceKey]
-
-	if !hasName && !hasNamespace {
-		return nil, nil
+func getSecretReference(secretParams secretParamsType, storageClassParams map[string]string, pvName string, pvc *v1.PersistentVolumeClaim) (*v1.SecretReference, error) {
+	nameTemplate, namespaceTemplate, err := getSecretNameAndNamespaceTemplate(secretParams, storageClassParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get name and namespace template from params: %v", err)
 	}
-
-	if len(nameTemplate) == 0 || len(namespaceTemplate) == 0 {
-		return nil, fmt.Errorf("%s and %s parameters must be specified together", nameKey, namespaceKey)
+	if nameTemplate == "" && namespaceTemplate == "" {
```

Comment:
you only need to check one empty string here, otherwise the previous function will return error.

Reply:
From my perspective it's easier to see what the check is doing this way